### PR TITLE
Change version to 0.0.1

### PR DESCRIPTION
**Pull request by dszmigielski**
**What**
Chang the version of tracer to 0.0.1

**Why**
We should not follow DataDog upstream versioning as this is separate solution. This resolves [Issue#141](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/141)

### CODE REVIEW

File: docs/RELEASING.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Releasing
+
+## Bumping version
+
+To change the version the following steps have to be executed:
+* Update the version number in [TracerVersion.cs](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/main/tools/Datadog.Core.Tools/TracerVersion.cs).
+* Run the [PrepareRelease tool](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/tree/main/build/tools/PrepareRelease): `dotnet run versions integrations`.
```
Comment (zacharycmontoya) [inline docs/RELEASING.md:7]:
```suggestion
* Run the [PrepareRelease tool](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/tree/main/build/tools/PrepareRelease): `dotnet run msi versions integrations`.
```

Using the `msi` argument is somewhat unrelated to "bumping version" but it updates the dll's and pdb's included in the MSI and must be done before releasing

**Comment by pellared** [comment]
This file has not been updated `src/WindowsInstaller/WindowsInstaller.wixproj` in line `20`:

```
    <OutputName>opentelemetry-dotnet-autoinstrumentation-1.26.1-$(Platform)</OutputName>
```

Maybe the PrepareRelease tool requires some changes?

File: docs/RELEASING.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# Releasing
```
Comment (pellared) [inline docs/RELEASING.md:1]: Would be good to add a point with:
```
Update [CHANGELOG.md](CHANGELOG.md) with new the new release.
```

File: docs/RELEASING.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+
+## Bumping version
```
[the remark this answers is not in the document]
Comment (dszmigielski) [reply docs/RELEASING.md:3]: I don't know if we want to rename that to "`Pre-Release`", this is instruction on bumping the version, it doesn't matter if it's pre-release or not. There is a flag for that in https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/main/tools/Datadog.Core.Tools/TracerVersion.cs

**Comment by zacharycmontoya** [comment]
> This file has not been updated `src/WindowsInstaller/WindowsInstaller.wixproj` in line `20`:
> 
> ```
>     <OutputName>opentelemetry-dotnet-autoinstrumentation-1.26.1-$(Platform)</OutputName>
> ```
> 
> Maybe the PrepareRelease tool requires some changes?

Yeah it looks like these lines need to be changed to recognize the new name: https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/b9d2c93e94a84c34f75de73950ab3f1b069ca47a/build/tools/PrepareRelease/SetAllVersions.cs#L180-L184

File: docs/RELEASING.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (pellared) [reply docs/RELEASING.md:3]: Pre-release usually means steps before you do the actual release. I imagine you always bump the version before releasing 😉
We can also do
```
## Pre-Release

### Bumping version
```

[the remark this answers is not in the document]
Comment (zacharycmontoya) [reply docs/RELEASING.md:3]: I think "Pre-Release" really means "Prepare Repository for Release" in that context

**Comment by pellared** [comment]
As noted during SIG, the procedure of updating the version may be helpful when pulling changes from upstream. It is worth to update `docs/PULL_FROM_UPSTREAM.md`

**Comment by pjanotti** [comment]
@dszmigielski another file that needs update: `samples\ServiceFabricRemoting\DatadogInstall.ps1`, notice the end of the line:
```ps1
if (Test-Path env:OTEL_TRACER_VERSION) { $OTEL_TRACER_VERSION = $env:OTEL_TRACER_VERSION } else { $OTEL_TRACER_VERSION = '1.24.0' }
```

It looks like this should be added to the tool too: default to the latest unless specified via env var.

**Comment by dszmigielski** [comment]
> @dszmigielski another file that needs update: `samples\ServiceFabricRemoting\DatadogInstall.ps1`, notice the end of the line:
> 
> ```powershell
> if (Test-Path env:OTEL_TRACER_VERSION) { $OTEL_TRACER_VERSION = $env:OTEL_TRACER_VERSION } else { $OTEL_TRACER_VERSION = '1.24.0' }
> ```
> 
> It looks like this should be added to the tool too: default to the latest unless specified via env var.

It seems this is missed in the upstream as well. There is still 1.24 there.

**Comment by dszmigielski** [comment]
> As noted during SIG, the procedure of updating the version may be helpful when pulling changes from upstream. It is worth to update `docs/PULL_FROM_UPSTREAM.md`

I will add point to update the version in the version file before running the `PrepareRelease` tool.

File: build/tools/PrepareRelease/publish-all.bat [hunk 4]
```diff
@@ -1,4 +1,4 @@
-
+WindowsInstaller.wixproj
```
Comment (pjanotti) [inline build/tools/PrepareRelease/publish-all.bat:1]: This seems wrong.

**Review by pjanotti — COMMENTED** [review]
> It seems this is missed in the upstream as well. There is still 1.24 there.

Yes, likely should be added to the tool anyway. BTW, the file should be renamed too.

@zacharycmontoya was the version of the Service Frabric PowerShell script intentionally left on 1.24? At first, it looks like something that should be covered by the tool.

File: docs/RELEASING.md (hunk 2 above)
Comment (pellared) [reply docs/RELEASING.md:1]: Addressed https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/155/commits/0fda82c2e501bbac2f1f734dc1e6b0a33e60c4d1

**Review by pellared — APPROVED** [review]

**Review by pellared — APPROVED** [review]

**Comment by zacharycmontoya** [comment]
> @zacharycmontoya was the version of the Service Frabric PowerShell script intentionally left on 1.24? At first, it looks like something that should be covered by the tool.

This is unintentional, it should be included in the version update. I'll update it on the Datadog side, but it's not a critical issue as it's not deployed anywhere, it's just a reference for consumers of the library.

**Review by zacharycmontoya — APPROVED** [review]

**Review by nrcventura — APPROVED** [review]

File: samples/AutomaticTraceIdInjection/Log4NetExample/Log4NetExample.csproj [hunk 5]
```diff
@@ -6,7 +6,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Datadog.Trace" Version="1.26.1" />
+    <PackageReference Include="Datadog.Trace" Version="0.0.1" />
```
Comment (lucaspimentel) [inline samples/AutomaticTraceIdInjection/Log4NetExample/Log4NetExample.csproj:9]: Probably not a blocker for this PR, but none of these "sample" projects will build with a NuGet reference to `Datadog.Trace` version 0.0.1 (such a packages does not exist).

**Pull request merged**